Re-upgrade Substrate and Gateway accounts for validators

jflatow opened the pull request:
WIP: CASH benchmarks aren't working for some reason (commented out) and test is failing due to another account minimal existence issue.

Also had to delete GrandpaRpcHandler, but hopefully not needed now?

UPDATE: Fixed the account minimal existence issues (and added back Gateway accounts), but benchmarks are still broken and scenarios aren't working yet either.

jflatow committed:
Re-upgrade Substrate

WIP: CASH benchmarks aren't working for some reason (commented out)
 and test is failing due to another account minimal existence issue

Also had to delete GrandpaRpcHandler, but hopefully not needed now?

hayesgm reviewed (approved):
Looks awesome; any idea how well the scens pass here?

jflatow committed:
Re-add Gateway addresses, which validators now have

Validators are now required to have a minimum CASH balance in order to set session keys, and therefore validate.
On committing a pipeline change, any account with a balance above the minimum is marked as existing in the Substrate account store.
Any account with a balance below the minumum is purged from the account store.

More test coverage needed.

jflatow commented:
> Looks awesome; any idea how well the scens pass here?

They don't yet 😅

jflatow committed:
Really use min balance path in tests

jflatow commented:
Subsumed by #353
